src/mpl2/src/SimulatedAnnealingCore.cpp

```diff
@@ -518,7 +518,10 @@ void SimulatedAnnealingCore<T>::exchangeMacros()
       neg_index2 = i;
     }
   }
-  std::swap(neg_seq_[neg_index1], neg_seq_[neg_index2]);
+
+  if (neg_index1 >= 0 && neg_index2 >= 0) {
```

Comment:
If this ever fails it would be a bug.

Reply:
Should I add an error message if it fails?

Reply:
That would be nice.  You can then move the swap out/after the if block.